Fixed potential bugs occurring under certain conditions

andirayo opened the pull request:
> Changed name of global function package_installed?() to r_package_installed?(), as, there are potential overlaps with other cookbooks, eg. function-name overlaps with NodeJS-cookbook provider NPM (which leads to an error when used in combination).
> Installing package (lib)readline(-dev) using readline-cookbook, as, R-installation when invoked without parameter "--with-readline=no" may fail oherwise.
> Preventing (potential) directory not found errors by creating /etc/-folder in installation-directory before using it.

andirayo commented:
I don't think that the failing Check (on Travis) has to do with my changes, as the error-output says:
'The command "rake" exited with 1.'
I tested the cookbook (and my changes) on both Ubuntu v12 and v14, as well as, on Amazon Linux v2014.09.
The bugs I encountered are fixed and the installation of R, as well as, R-packages works fine

stevendanna commented on recipes/default.rb:
```diff
@@ -21,12 +21,25 @@
 
+# installs package (lib)readline(-dev), which is necessary for (output of) R-installation (ie. configure-step)
+#   another option is to install/configure with parameter "--with-readline=no"
+include_recipe 'readline'
```
I think this should probably go in `recipes/install_source.rb`

stevendanna commented on metadata.rb:
```diff
@@ -9,6 +9,7 @@
 depends "apt"
 depends "ark"
 depends "build-essential"
+depends 'readline'          , '~> 0.0.3'
```
Is there a reason for the version constraint here?

stevendanna commented:
@andirayo Thanks for the contribution.  I agree that the travis failures are unrelated.  I've left a few comments.  Also, do you mind rebasing this branch to clean up some of the commit history?

andirayo commented:
@stevendanna: By rebasing my branch, do you want me to consolidate my commits into 1 commit (as I think, there is no actual rebasing between out branches)?

As for your diff-comments:
- As suggested by you, I moved the inclusion "readline" into recipe install_source.
- I removed the version-dependency. I originally added it, as, the "readline" dependency is not over-important for this cookbook (future updates are probably not needed) and I wanted to prevent it from breaking the cookbook in the future.

stevendanna commented:
@andirayo It doesn't necessarily need to be one commit.  I'd prefer 1 commit for each logical change:

- addition of readline 
- rename of package_installed? to r_package_installed?
- creation of /etc directory
- addition of .idea to the gitignore

I can always do this at merge time if you don't have time.

andirayo commented:
@stevendanna: Ok, I got it, now. :)
I rebased my commits to match the form "1 commit for each logical change"

andirayo commented:
@stevendanna: ping

stevendanna commented:
@andirayo This has been merged.  I'd like to add some testing before doing another release to the community site, but I won't let this sit unreleased for too long.  Thanks!

andirayo commented:
Cool thanks, I will keep an eye on it! :)
